src/net/atlas/db.rs

```diff
@@ -113,9 +115,32 @@ impl AtlasDB {
         Ok(())
     }
 
+    pub fn should_keep_attachment(
+        &self,
+        contract_id: &QualifiedContractIdentifier,
+        attachment: &Attachment,
+    ) -> bool {
+        if self.atlas_config.contracts.contains(contract_id) {
```

Comment:
Shouldn't this be `if !self.atlas_config.contracts.contains(contract_id) {`?

Reply:
Oups, I've been re-writing this condition a few times - will fix and test.

Reply:
Fixed in test: add coverage for test_keep_uninstantiated_attachments